src/sticker-builder/stickers/insta-sticker.ts

```diff
@@ -0,0 +1,86 @@
+import { snakeCase } from 'lodash';
+import { classToPlain } from 'class-transformer';
+
+export abstract class InstaSticker {
+  public width: number;
+  public height: number;
+  public rotation: number = 0.0;
+  public x: number = 0.0;
+  public y: number = 0.0;
+  public z: number = 0;
+
+  public isSticker: boolean;
+
+  public abstract get id(): string;
+
+  public abstract get key(): string;
+
+  protected constructor(width: number, height: number, isSticker: boolean = true) {
+    this.width = width;
+    this.height = height;
+    this.isSticker = isSticker;
+  }
+
+  public center(): this {
+    this.x = 0.5;
+    this.y = 0.5;
+    return this;
+  }
+
+  public rotateDeg(deg: number): this {
+    this.rotation = deg % 360.0;
+    return this;
+  }
+
+  public scale(factor: number): this {
+    this.width *= factor;
+    this.height *= factor;
+    return this;
+  }
+
+  public moveForward(layers: number = 1): this {
+    this.z += layers;
+    return this;
+  }
+
+  public moveBackwards(layers: number = 1): this {
+    return this.moveForward(-layers);
+  }
+
+  public right(): this {
+    this.x = 1.0 - this.width / 2;
+    return this;
+  }
+
+  public left(): this {
+    this.x = this.width / 2;
+    return this;
+  }
+
+  public top(): this {
+    this.y = this.height / 2;
+    return this;
+  }
+
+  public bottom(): this {
+    this.y = 1.0 - this.height / 2;
+    return this;
+  }
+
+  protected toSnakeCase(obj: object): object {
+    const result = {};
+    const pairs = Object.entries(obj);
+    for (const [key, value] of pairs) {
+      if (Array.isArray(value)) {
+        Object.defineProperty(result, snakeCase(key), { value: value.map(x => this.toSnakeCase(x)), enumerable: true });
+      } else {
+        Object.defineProperty(result, snakeCase(key), { value, enumerable: true });
+      }
+    }
+    return result;
+  }
+
+  public toJSON() {
+    return this.toSnakeCase(classToPlain(this));
```

Comment:
Ooook, so much less code 😄 

One more critical notice. As we all know - instagram is unpredictable. We don't have docs to their API so we cant be sure that automatically change the case will fit forever. In addition this approach is unobvious and has some pitfalls.

Let's pretend we need to send key like `sticker_version2`
Surprise :) you can't do it

```js
_.snakeCase('stickerVersion2') // sticker_version_2
```

Instagram can even make some keys camelCase. So it's unstable to automatically convert case.

But if you decide to keep it unchanged - let's at least use special library for that, because `this.toSnakeCase` looks like **SRP** violation

https://www.npmjs.com/package/snakecase-keys

Reply:
This is true, but Instagram uses only snake-case in its own api (only in the graph apis they use camel-case).